pkg/reconciler/v1alpha1/service/service.go

```diff
@@ -248,17 +248,39 @@ func (c *Reconciler) createConfiguration(service *v1alpha1.Service) (*v1alpha1.C
 	return c.ServingClientSet.ServingV1alpha1().Configurations(service.Namespace).Create(cfg)
 }
 
+func configSemanticEquals(config, desiredConfig *v1alpha1.Configuration) bool {
+	return equality.Semantic.DeepEqual(desiredConfig.Spec, config.Spec) &&
+		equality.Semantic.DeepEqual(desiredConfig.ObjectMeta.Labels, config.ObjectMeta.Labels)
+}
+
+// mergeRouteLabel sets desiredConfig[serving.RouteLabelKey] to same
+// as config[serving.RouteLabelKey], so that we do nothing about
+// configuration label serving.RouteLabelKey in our reconciliation.
+func mergeRouteLabel(desiredConfig, config *v1alpha1.Configuration) {
+	routeLabel, existed := config.ObjectMeta.Labels[serving.RouteLabelKey]
+	if !existed {
+		delete(desiredConfig.ObjectMeta.Labels, serving.RouteLabelKey)
+	} else {
+		desiredConfig.ObjectMeta.Labels[serving.RouteLabelKey] = routeLabel
+	}
+}
+
 func (c *Reconciler) reconcileConfiguration(ctx context.Context, service *v1alpha1.Service, config *v1alpha1.Configuration) (*v1alpha1.Configuration, error) {
 	logger := logging.FromContext(ctx)
 	desiredConfig, err := resources.MakeConfiguration(service)
 	if err != nil {
 		return nil, err
 	}
+	// Route label is automatically set by another reconciler.  We
+	// want to ignore that label in our reconciliation here by setting
+	// desiredConfig[serving.RouteLabelKey] to the same as
+	// config[serving.RouteLabelKey].
+	mergeRouteLabel(desiredConfig, config)
 
 	// TODO(#642): Remove this (needed to avoid continuous updates)
 	desiredConfig.Spec.Generation = config.Spec.Generation
 
-	if equality.Semantic.DeepEqual(desiredConfig.Spec, config.Spec) {
+	if configSemanticEquals(config, desiredConfig) {
```

Comment:
nit: the argument list is backwards from all of the other methods. It would be nice to be consistent here.

Reply:
Done